sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -2150,6 +2150,15 @@ pi_result piDeviceGetInfo(pi_device Device, pi_device_info ParamName,
     // std::array<std::byte, 16>. For details about this extension,
     // see sycl/doc/extensions/IntelGPU/IntelGPUDeviceInfo.md.
     return ReturnValue(Device->ZeDeviceProperties->uuid.id);
+  case PI_DEVICE_INFO_ATOMIC_64: {
+    if (Device->ZeDeviceModuleProperties->flags &
+        ZE_DEVICE_MODULE_FLAG_INT64_ATOMICS) {
+      bool result = true;
+      std::memcpy(ParamValue, &result, sizeof(bool));
+      return PI_SUCCESS;
+    }
+    return PI_INVALID_VALUE;
+  }
```

Comment:
```suggestion
  case PI_DEVICE_INFO_ATOMIC_64:
    return ReturnValue(pi_bool{Device->ZeDeviceModuleProperties->flags &
        ZE_DEVICE_MODULE_FLAG_INT64_ATOMICS});
```

Reply:
And also remove this specialization: https://github.com/intel/llvm/blob/ec9d089684ef402ef12697bea094c60b3c35947a/sycl/source/detail/device_info.hpp#L237

Reply:
The specialization has been removed, but your suggestion for L0 case doesn't contains memcpy, therefore I modernize it a bit (pi_bool works weird with memcpy, so I copy the bool value), what do you think?

Reply:
The "copy" is in the ReturnValue helper, just look for code around that returns a boolean info already

Reply:
Good hint! (but still, if I return pi_bool then device.has(aspect::atomic64) doesn't output correct answer) Also, due to failed tests, it looks like the hip backend used the deleted specialization for cuda. I think, we should undo the removal of specialization, what do you think about this?

Reply:
> if I return pi_bool then device.has(aspect::atomic64) doesn't output correct answer

I bet this is because `pi_bool` is 4 bytes, so either change to `bool`, or write `pi_bool{(Device->ZeDeviceModuleProperties->flags & ZE_DEVICE_MODULE_FLAG_INT64_ATOMICS) != 0}`

> all backends used the deleted specialization for cuda

I don't understand what do you mean by "all backends for cuda" and I don't see CUDA backend using that specialization, i.e. it always returns a boolean and PI_SUCCESS. So please continue to remove the specialization

Reply:
> I don't understand what do you mean by "all backends for cuda"
I mean this specialization is called on any backend if we write device.has(aspect::atomic64).

Reply:
Specialization deleted.